File: Top Interview 150/Java Code/Stack/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        int n=s.length();
        Stack<Character>st=new Stack<>(); 
        for(int i=0;i<n;i++){
            if(st.size()==0) st.push(s.charAt(i));
                else if(s.charAt(i)=='(' || s.charAt(i)=='{' || s.charAt(i)=='[') st.push(s.charAt(i));
               else if(s.charAt(i)==')' && st.peek()=='(') st.pop();
               else if(s.charAt(i)=='}' && st.peek()=='{') st.pop();
               else if(s.charAt(i)==']' && st.peek()=='[') st.pop();
              else return false;
            
            }
        return st.size()==0;
    }
}

File: Top Interview 150/C++ Code/Stack/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(int i=0;i<s.length();i++){
            if(st.size()==0) st.push(s[i]);
           else if(s[i]=='(' || s[i]=='{'|| s[i]=='[') st.push(s[i]);
            else if(st.top()=='(' && s[i]==')') st.pop();
            else if(st.top()=='{' && s[i]=='}') st.pop();
            else if(st.top()=='[' && s[i]==']') st.pop();
            else return false;

            
        }
        return st.size()==0;
    }
};
